client/finality-grandpa/src/finality_proof.rs

```diff
@@ -314,11 +314,10 @@ pub(crate) fn prove_finality<Block: BlockT<Hash=H256>, B: BlockchainBackend<Bloc
 
 		if let Some(justification) = blockchain.justification(current_id)? {
 			// check if the current block enacts new GRANDPA authorities set
-			let parent_id = BlockId::Number(current_number - One::one());
-			let new_authorities = authorities_provider.authorities(&parent_id)?;
+			let new_authorities = authorities_provider.authorities(&current_id)?;
 			let new_authorities_proof = if current_authorities != new_authorities {
 				current_authorities = new_authorities;
```

Comment:
Shouldn't `authorities_set_id` also get incremented?

Reply:
Nope - `authorities_set_id` are passed by the requester (who wants finality proof) and are only used to check that we have the same set (to fail if there have been forced changes).